packages/terra-status-view/src/StatusView.jsx

```diff
@@ -0,0 +1,211 @@
+      componentsHeight += this.buttonsNode.offsetHeight;
+    }
+    if (this.messageNode) {
+      componentsHeight += this.messageNode.offsetHeight;
+    }
+    if (this.dividerNode) {
+      componentsHeight += this.dividerNode.offsetHeight;
+    }
+
+    if (this.glyphNode && viewHeight >= componentsHeight + this.glyphNode.offsetHeight) {
+      // if have a glyph to show and can fit inside the containers height, distribute remaining padding to center
+      showGlyph = true;
+      if (!this.props.isAlignedTop) {
+        const remainingHeight = viewHeight - (componentsHeight + this.glyphNode.offsetHeight);
+        paddingTop = remainingHeight * 0.4;
+        paddingBottom = remainingHeight * 0.6;
+      }
+    } else if (!this.props.isAlignedTop) {
+      // glyph does not exist or does not fit, distribute remaining padding if any to center
+      const remainingHeight = viewHeight - componentsHeight;
+      if (remainingHeight > 0) {
+        paddingTop = remainingHeight * 0.4;
+        paddingBottom = remainingHeight * 0.6;
+      }
+    }
+
+    // set which components can fit into the view and add the additional 20px
+    // top and bottom padding to the remaining additional padding leftover
+    this.setState({
+      showGlyph,
+      paddingTop,
+      paddingBottom,
+    });
+  }
+
+  render() {
+    const {
+    children,
+    isAlignedTop,
+    isGlyphHidden,
+    message,
+    title,
+    variant,
+    ...customProps
+    } = this.props;
+
+    const { intl } = this.context;
+
+    let glyphSection;
+    if (!isGlyphHidden && this.state.showGlyph) {
+      glyphSection = <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}><svg className={cx(variant)} /></div>;
+    }
+    let messageSection;
+    if (message) {
+      messageSection = <div className={cx('message')} ref={(element) => { this.messageNode = element; }}>{message}</div>;
+    }
+    let buttonSection;
+    if (children.length) {
+      buttonSection = <div className={cx('buttons')} ref={(element) => { this.buttonsNode = element; }}>{children}</div>;
+    }
+    let defaultTitle = title;
+    if (!defaultTitle) {
+      defaultTitle = (variant === StatusView.variants.CUSTOM) ? '' : intl.formatMessage({ id: `Terra.status-view.${variant}` });
+    }
+    let dividerSection;
+    if (messageSection || buttonSection) {
+      dividerSection = <div className={cx('divider')} ref={(element) => { this.dividerNode = element; }}><Divider /></div>;
+    }
+
+    const titleSection = <div className={cx('title')} ref={(element) => { this.titleNode = element; }}>{defaultTitle}</div>;
+
+    const statusViewClassNames = cx([
+      'status-view',
+      customProps.className,
+    ]);
+
+    const statusViewInnerStyles = {
+      paddingTop: `${this.state.paddingTop}px`,
+      paddingBottom: `${this.state.paddingBottom}px`,
+    };
+
+    return (
+      <div
+        {...customProps}
+        className={statusViewClassNames}
+      >
```

Comment:
Why are two wrapping divs needed?

Reply:
the outer wrapping will apply the default padding around content alway, the content is inside its own div to have the remaining vertical space in the view distributed if there is any